### ISSUE TITLE
Bug: Renaming of executables breaks tutorials

### ISSUE THREAD

**Issue by gpacix** [ggerganov/llama.cpp#8397]
### What happened?

I was getting started with llama.cpp and the existing tutorials all say to run `./main` and`./server`, which have very recently (1c641e6a, 2024-06-12) been renamed to `./llama-cli` and `./llama-server` (as I understand it).

Could you please (temporarily) restore the old names, and deprecate them instead— by adding shell scripts with the old names that announce the deprecation and run the new executable with the same arguments?  Even just announcing the new name and quitting would be helpful.

While I eventually figured this out, it is a frustrating stumbling block for new users (and possibly existing ones, as well).

### Name and Version

```
./llama-cli --version
version: 0 (unknown)
built with Apple clang version 13.0.0 (clang-1300.0.29.30) for x86_64-apple-darwin20.6.0
```
Note: I downloaded the source, version `b3334`, and built it myself.

### What operating system are you seeing the problem on?

Mac

### Relevant log output

```shell
$ ./main
bash: ./main: No such file or directory
$ ./server
bash: ./server: No such file or directory
```

**Comment by dspasyuk** [comment]
There is no going back to the old names. Some manuals tutorials have been updated already, you can find them here: https://github.com/ggerganov/llama.cpp/blob/master/examples/main/README.md and here https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md They are located at the end of README.

**Comment by gpacix** [comment]
Yes, I understand that, that's fine.

All I am asking for is a reasonable, useful error message when someone tries to run `./main`​, along the lines of:

"Error: ./main has been renamed to ./llama-cli; please update your scripts and documentation"

(and similarly for `./server`).

**Comment by slaren** [comment]
For the most part this should have been addressed in #8283

**Comment by gpacix** [comment]
A project that fixes problems by the time I can even report them? Nice! Thanks!

**Comment by HanClinto** [comment]
> For the most part this should have been addressed in #8283

One change we made in #8283 was to make it so that it only builds the replacement `main` binary (that gives the name-change warning message) if it sees an existing `main` binary in the same folder.

```
(base) ➜  llama.cpp git:(legacy-binaries) ✗ ./main

WARNING: The binary 'main' is deprecated.
 Please use 'llama-cli' instead.
 See https://github.com/ggerganov/llama.cpp/tree/master/examples/deprecation-warning/README.md for more information.
```

However, because it doesn't always build the binaries, then this won't help for people who are following a tutorial and typing in `main` in a fresh install -- for that, we might want to remove the `main` binary filename check and ensure that at least that binary builds all the time (for now, until everything else gets upgraded). I think that would address the majority of what @gpacix is asking for?

**Comment by HanClinto** [comment]
> All I am asking for is a reasonable, useful error message when someone tries to run `./main`​, along the lines of:
> 
> "Error: ./main has been renamed to ./llama-cli; please update your scripts and documentation"

I just opened a PR (#8404) to build the name-migration-alert `main` binary every time `make` is run. Hopefully this will help some of your use-case, @gpacix?

> (and similarly for `./server`).

Are there tutorials that you've seen that instruct people to run `server`? If so, then I'll modify that PR to also build `server` every time as well.
